src/azstoragetorch/_client.py

```diff
@@ -182,6 +237,18 @@ def _get_max_in_flight_requests(self) -> int:
         cpu_count_fn = getattr(os, "process_cpu_count", os.cpu_count)
         return min(32, (cpu_count_fn() or 1) + 4)
 
+    def _get_executor(self) -> concurrent.futures.Executor:
+        # We want executor creation to be lazy instead of instantiating immediately in
+        # the constructor because the executor itself is not pickleable. This is an issue
+        # when workers are used by PyTorch's DataLoader as workers are spawned as processes.
+        # So we delay executor creation until it is needed for reading/writing data which will
+        # happen after processes are spawned.
+        if self._executor is None:
```

Comment:
Any issues with concurrency here? I assume only a single thread will ever be calling this for each class?

Reply:
There should not be an issue based on how the client is used today. Specifically, these methods should only be called from a single thread because the public interfaces that use the method (e.g. `BlobIO`) should only ever be handled by a single thread. If customers are using multiple threads (which is not an expected workflow), they will need to make sure to be protecting access to the `BlobIO` with a lock.